contents/data/sql/SQL.md

```diff
@@ -0,0 +1,151 @@
+
+---
+
+### How does SQL work?
+
+SQL uses **queries** to retrieve data. Here is an example of how an SQL query is used.
+
+Suppose we have the data table `Students` shown here:
+
+| ID | Name  | Course | Faculty |
+| -- | ----- | ------ | ------- |
+| 1  | Alex  | CS202  | CS      |
+| 2  | Bob   | MA303  | MA      |
+| 3  | Cathy | CS202  | CS      |
+| 4  | Daren | CS202  | CS      |
+| 5  | Ellie | CS101  | CS      |
+| 6  | Fred  | MA303  | MA      |
+| 7  | Gary  | CS101  | CS      |
+| 8  | Henry | CS404  | CS      |
+
+We can use this SQL query to retrieve information from this table:
+```sql
+SELECT Course, COUNT(*) num
+FROM Students
+WHERE Faculty = 'CS'
+GROUP BY Course
+HAVING COUNT(*) > 1
+ORDER BY num;
+```
+
+This query first filters the entries in the `Students` table such that only entries that have `CS` as the faculty are considered.
+After that, it groups those entries into the 3 courses: `CS101`, `CS202` and `CS404`.
+Then, it removes courses that do not have more than 1 student, i.e., `CS404` is removed from consideration.
+Finally, it returns a list of courses with a count of the number of students, ordered in increasing order.
+So, the output of the query is:
+
+| Course | num |
+| ------ | --- |
+| CS101  | 2   |
+| CS202  | 3   |
+
+You can see how this simple query can prove to be extremely useful for getting this information when the table has tens of entries (or many more). Apart from retrieving information, SQL can also be used for creating, deleting and manipulating data with queries like `INSERT`, `DELETE` and `UPDATE` for entries, in addition to `CREATE`, `DROP` and `ALTER` for tables as a whole.
+
+For example, you can set up the structure of the table shown in Table 1 by executing the `CREATE TABLE` query shown here.
+It creates a new data table `Students`, with 4 fields (`ID`, `Name`, `Course` and `Faculty`) and specifies their data types.
+
+```sql
+CREATE TABLE Students(ID int, Name varchar(255), Course varchar(255), Faculty varchar(255));
+```
+
+We can also remove rows containing the `MA` faculty from Table 1 with this `DELETE` query.
+
+```sql
+DELETE
+FROM Students
+WHERE Faculty='MA';
+```
+
+<box type="info">
+  You can experiment with <a href="https://www.db-fiddle.com/f/kHqV2edUGxCc1dU6vE6CmS/1">this example on DB Fiddle</a> (shown in Figure 1 below) by entering SQL queries and then running them.
+</box>
+
+<center>
+<img src="DBFiddle.png" alt="Running queries on DB Fiddle" width="90%">  
+
+_Figure 1. Using DB Fiddle to run queries in MySQL_
+</center>
+
+**Online editors** like DB Fiddle can be used while learning SQL. However, to use SQL in application development, you need to consider other alternatives for running SQL queries:
+
+- **RDBMS software**: RDBMS installations usually come with a <tooltip content="Graphical User Interface">GUI</tooltip> (see Figure 2), but can also be accessed from the command line (see Figure 3).
```

Comment:
Feels like the images below can be made a bit larger so that the text in their matches the text size of the main body of the article.